### PR TITLE
NoJira - Add demo preview deeplink handler

### PR DESCRIPTION
**Pull request by thomson-t**
### Background ###

* Capture the preview parameter from the URL that app is launched with
* Handle both app running in background and launched fresh.
* Extract the `config` query string and set it in the AppState observable object
* Navigate to the preview page from home when environment variable is set
* Pass the preview JSON to the viewModel and trigger the preview

### What Has Changed: ###

Handle deeplinks with custom URI scheme `rokt://`.

### How Has This Been Tested? ###

Tested locally

https://github.com/user-attachments/assets/a6aabef3-7e2f-4b0f-90cd-a5d176572d7c

```bash
xcrun simctl openurl booted "rokt://demo/preview?config={\"tagId\":\"3198447216177237634\",\"previewId\":\"3348516972682149889\",\"versionId\":\"1726322162136\",\"creativeIds\":[\"3335250570341581807\",\"3334286925719734182\",\"3341268103002062853\",\"3341270564018389003\"],\"language\":\"en\",\"layoutVariantIds\":[\"3348516972682149890\",\"3348516972682149891\"]}"
```

### Notes

Add any notes or extra information here that might be useful to the reviewer if applicable i.e. links to documentation/blogs or dashboards.

### Checklist: ###

- [x] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [x] I have verified that CI completes successfully.
- [x] All insignificant commits have been squashed.

### CODE REVIEW

**Review by danial-rokt — APPROVED** [review]

**Comment by hehoo** [comment]
Can we do a similar thing in Android?

File: RoktDemo/Resources/SceneDelegate.swift [hunk 1]
```diff
@@ -64,6 +77,14 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
         // to restore the scene back to its current state.
     }
 
+    private func handleURLContexts(_ urlContexts: Set<UIOpenURLContext>) {
+        if let urlContext = urlContexts.first {
+            let url = urlContext.url
+            let queryItems = URLComponents(url: url, resolvingAgainstBaseURL: false)?.queryItems
+            let config = queryItems?.first(where: { $0.name == "config" })?.value
```
Comment (hehoo) [inline RoktDemo/Resources/SceneDelegate.swift:84]: Do we want to validate the config a bit here?

**Review by hehoo — APPROVED** [review]
Good job! @thomson-t

File: RoktDemo/Resources/SceneDelegate.swift (hunk 1 above)
Comment (thomson-t) [reply RoktDemo/Resources/SceneDelegate.swift:84]: IMO that can be done on the layout preview page. The URL handler simply get the config query param and pass it to the relevant page that can handle it. That page would know the domain model and the validations needed.

**Comment by thomson-t** [comment]
> Can we do a similar thing in Android?

It is already available in Android https://github.com/ROKT/rokt-demo-android/pull/82
Since it does not affect any existing functionalities, we took the liberty to merge it.

**Review by jamesnrokt — APPROVED** [review]

**Pull request merged**